### PR TITLE
Create wedding tests

### PR DESCRIPTION
**Pull request by SherniceSng**
Added tests for ClientTest.java, NameMatchesWeddingPredicateTest.java, UniqueWeddingListTest.java

Added new exceptions for Wedding class: DuplicateWeddingException.java and WeddingNotFoundException.java

Note: One of the tests in UniqueWeddingListTest.java fails, will fix it soon!

### CODE REVIEW

File: src/test/java/seedu/address/model/wedding/ClientTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        assertFalse(Client.isValidClientIndex(test));
+
+        test = "";
+        assertFalse(Client.isValidClientIndex(test));
+
+        test = "  123   ";
+        assertFalse(Client.isValidClientIndex(test));
+
+        test = "123a";
+        assertFalse(Client.isValidClientIndex(test));
```
Comment (jowhee3011) [inline src/test/java/seedu/address/model/wedding/ClientTest.java:86]: Good consideration

File: src/test/java/seedu/address/model/wedding/NameMatchesWeddingPredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        predicate = new NameMatchesWeddingPredicate(Collections.singletonList("Alice"));
+        assertTrue(predicate.test(new WeddingBuilder().withName("Bob Alice Wedding").build()));
+
+        predicate = new NameMatchesWeddingPredicate(Arrays.asList("Alice", "Bob"));
+        assertTrue(predicate.test(new WeddingBuilder().withName("Alice Bob Wedding").build()));
+
+        predicate = new NameMatchesWeddingPredicate(Arrays.asList("Alice", "Bob"));
+        assertTrue(predicate.test(new WeddingBuilder().withName("Alice Bob Tan Wedding").build()));
+
+        // Mixed-case keywords
```
Comment (jowhee3011) [inline src/test/java/seedu/address/model/wedding/NameMatchesWeddingPredicateTest.java:58]: Good consideration

**Review by jowhee3011 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/wedding/exceptions/DuplicateWeddingException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package seedu.address.model.wedding.exceptions;
+
+/**
+ * Signals that the operation will result in duplicate Wedding (Wedding are considered duplicates if they have the same
+ * name, venue, date and client).
+ */
+public class DuplicateWeddingException extends RuntimeException {
+    public DuplicateWeddingException() {
+        super("Operation would result in duplicate persons");
```
Comment (VedJoshi) [inline src/main/java/seedu/address/model/wedding/exceptions/DuplicateWeddingException.java:9]: specific errors being thrown, nice.

**Review by VedJoshi — APPROVED** [review]

**Pull request merged**